cygnus-arcgis/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIArcgisFeatureTableSink.java

```diff
@@ -0,0 +1,789 @@
+
+        switch (attrType) {
+
+            case "geo:json":
+                try {
+                    JsonObject location = attrValue.getAsJsonObject();
+                    if (location.get("type").getAsString().equals("Point")) {
+                        JsonArray coordinates = location.get("coordinates").getAsJsonArray();
+                        double latitude = coordinates.get(0).getAsDouble();
+                        double longitude = coordinates.get(1).getAsDouble();
+
+                        Point point = new Point(latitude, longitude);
+                        feature.setGeometry(point);
+
+                    } else {
+                        LOGGER.warn("Invalid geo:json type, only points allowed: " + location.toString());
+                    }
+                } catch (Exception e) {
+                    LOGGER.error("Invalid geo:json format, (sikipped): " + attrValue.toString() + " - Error: "
+                            + e.getMessage());
+                }
+                break;
+            case "boolean":
+                if (attrValue.isJsonPrimitive() && attrValue.getAsJsonPrimitive().isBoolean()) {
+                    feature.addAttribute(attrName, attrValue.getAsBoolean());
+                } else {
+                    String strValue = attrValue.getAsString().toLowerCase().trim();
+                    boolean result = "true".equals(strValue) || "1".equals(strValue);
+                    feature.addAttribute(attrName, result);
+                }
+                break;
+            case "datetime":
+                String dateStr = attrValue.toString();
+                feature.addAttribute(attrName, parseFiwareDate(dateStr));
+                break;
+
+            default:
+                // Verify if it is a string (it is into quotation marks)
+                if (isQuoted(attrValue.toString())) {
+                    // Insert unquoted
+                    String strValue = URLDecoder.decode(unquote(attrValue.toString()));
+                    feature.getAttributes().put(attrName, strValue);
+                } else {
+                    try {
+                        // Try to insert as Integer
+                        feature.addAttribute(attrName, Integer.parseInt(attrValue.toString()));
+                    } catch (NumberFormatException e2) {
+                        try {
+                            // Try to insert as Double
+                            feature.addAttribute(attrName, Double.parseDouble(attrValue.toString()));
+                        } catch (NumberFormatException e3) {
+                            // If all fails, insert as String
+                            LOGGER.warn(
+                                    "[NGSIArcgisAggregator] Unquoted String attribute: " + attrName + ":" + attrValue);
+                            String strValue = URLDecoder.decode(attrValue.toString());
+                            feature.addAttribute(attrName, strValue);
```

Comment:
This seems to be kind of tricky. Maybe this will be throwing a lot of exceptions and it would affect the performance.

Can this be done using native gson validations?

`                    if (attrValue.isJsonPrimitive()){
                        if (attrValue.getAsJsonPrimitive().isNumber()))
                        try {
                            feature.addAttribute(attrName, attrValue.getAsInt());
                        }

                    ...
`

Reply:
You still need the second try/catch block to control errors while parsing Double values.

The point is that JsonElement.isNumber() doesn't distinguish between integer and double values.
isJsonPrimitive and isNumber were removed for performance reasons since they make any difference.